icu4c/source/i18n/number_usageprefs.cpp

```diff
@@ -107,37 +107,44 @@ void Usage::set(StringPiece value) {
 // measures.
 void mixedMeasuresToMicros(const MaybeStackVector<Measure> &measures, DecimalQuantity *quantity,
                            MicroProps *micros, UErrorCode status) {
-    micros->mixedMeasuresCount = measures.length() - 1;
-    if (micros->mixedMeasuresCount > 0) {
-#ifdef U_DEBUG
-        U_ASSERT(micros->outputUnit.getComplexity(status) == UMEASURE_UNIT_MIXED);
-        U_ASSERT(U_SUCCESS(status));
-        // Check that we received measurements with the expected MeasureUnits:
-        MeasureUnitImpl temp;
-        const MeasureUnitImpl& impl = MeasureUnitImpl::forMeasureUnit(micros->outputUnit, temp, status);
-        U_ASSERT(U_SUCCESS(status));
-        U_ASSERT(measures.length() == impl.singleUnits.length());
-        for (int32_t i = 0; i < measures.length(); i++) {
-            U_ASSERT(measures[i]->getUnit() == impl.singleUnits[i]->build(status));
+    micros->mixedMeasuresCount = measures.length();
+
+    // Mixed units: except for the last value, we pass all values to the
+    // LongNameHandler via micros->mixedMeasures.
+    if (micros->mixedMeasures.getCapacity() < micros->mixedMeasuresCount) {
+        if (micros->mixedMeasures.resize(micros->mixedMeasuresCount) == nullptr) {
+            status = U_MEMORY_ALLOCATION_ERROR;
+            return;
         }
-        (void)impl;
-#endif
-        // Mixed units: except for the last value, we pass all values to the
-        // LongNameHandler via micros->mixedMeasures.
-        if (micros->mixedMeasures.getCapacity() < micros->mixedMeasuresCount) {
-            if (micros->mixedMeasures.resize(micros->mixedMeasuresCount) == nullptr) {
-                status = U_MEMORY_ALLOCATION_ERROR;
-                return;
-            }
-        }
-        for (int32_t i = 0; i < micros->mixedMeasuresCount; i++) {
+    }
+
+    for (int32_t i = 0; i < micros->mixedMeasuresCount; i++) {
+        switch (measures[i]->getNumber().getType()) {
+        case Formattable::kInt64:
             micros->mixedMeasures[i] = measures[i]->getNumber().getInt64();
+            break;
+
+        // NOTE: Quantity is the only double valued measure
```

Comment:
Optional: I think this could become an assert, e.g.:

U_ASSERT(indexOfQuantity < 0)

thereby proving that indexOfQuantity gets assigned only once in this loop?